Integrations/src/main/java/io/deephaven/integrations/python/PythonMergedListenerAdapter.java

```diff
@@ -0,0 +1,97 @@
+ * dependencies have been satisfied. The listener then invokes the Python listener object.
+ *
+ * The Python listener object must be a Python MergedListener instance that provides a "_process" method implementation
+ * with no argument.
+ */
+public class PythonMergedListenerAdapter extends MergedListener {
+    private final PyObject pyCallable;
+
+    /**
+     * Create a Python merged listener.
+     *
+     * @param recorders The listener recorders to which this listener will subscribe.
+     * @param dependencies The tables that must be satisfied before this listener is executed.
+     * @param listenerDescription A description for the UpdatePerformanceTracker to append to its entry description, may
+     *        be null.
+     * @param pyObjectIn Python listener object.
+     */
+    private PythonMergedListenerAdapter(
+            @NotNull ListenerRecorder[] recorders,
+            @Nullable NotificationQueue.Dependency[] dependencies,
+            @Nullable String listenerDescription,
+            @NotNull PyObject pyObjectIn) {
+        super(Arrays.asList(recorders), Arrays.asList(dependencies), listenerDescription, null);
+        Arrays.stream(recorders).forEach(rec -> rec.setMergedListener(this));
+        this.pyCallable = PythonUtils.pyMergeListenerFunc(pyObjectIn);
+    }
+
+    public static PythonMergedListenerAdapter create(
+            @NotNull ListenerRecorder[] recorders,
+            @Nullable NotificationQueue.Dependency[] dependencies,
+            @Nullable String listenerDescription,
+            @NotNull PyObject pyObjectIn) {
+        if (recorders.length < 2) {
+            throw new IllegalArgumentException("At least two listener recorders must be provided");
+        }
+
+        final NotificationQueue.Dependency[] allItems =
+                Stream.concat(Arrays.stream(recorders), Arrays.stream(dependencies))
+                        .filter(Objects::nonNull)
+                        .toArray(NotificationQueue.Dependency[]::new);
+
+        final UpdateGraph updateGraph = allItems[0].getUpdateGraph(allItems);
+
+        try (final SafeCloseable ignored = ExecutionContext.getContext().withUpdateGraph(updateGraph).open()) {
+            return new PythonMergedListenerAdapter(recorders, dependencies, listenerDescription, pyObjectIn);
+        }
+    }
+
+    public void replay() {
+        final RowSet emptyRowSet = RowSetFactory.empty();
+        final RowSetShiftData emptyShift = RowSetShiftData.EMPTY;
+        final ModifiedColumnSet emptyColumnSet = ModifiedColumnSet.EMPTY;
+
+        ArrayList<TableUpdate> updates = new ArrayList<>();
+        for (ListenerRecorder recorder : getRecorders()) {
+            final TableUpdate update =
+                    new TableUpdateImpl(recorder.getParent().getRowSet(), emptyRowSet, emptyRowSet, emptyShift,
+                            emptyColumnSet);
+            updates.add(update);
+        }
+
+        pyCallable.call("__call__", updates);
```

Comment:
This is not properly managing `RowSet` ownership/lifecycles.
```suggestion
        final ArrayList<TableUpdate> updates = new ArrayList<>();
        try {
            for (ListenerRecorder recorder : getRecorders()) {
                final TableUpdate update = new TableUpdateImpl(
                        recorder.getParent().getRowSet().copy(),
                        RowSetFactory.empty(),
                        RowSetFactory.empty(),
                        RowSetShiftData.EMPTY,
                        ModifiedColumnSet.EMPTY);
                updates.add(update);
            }
            pyCallable.call("__call__", updates);
        } finally {
            updates.forEach(TableUpdate::release);
        }
```